File: ANGEN/SampleFramework.Controls/trunk/Samples/Tabs Category/Firefox TabControl/VB/FirefoxTabControl/MainForm.vb
Imports FirefoxTabControl.My.Resources
Imports System.Globalization

Public Class MainForm
    Private Sub AddTabPage()
        _tabPageCount += 1
        _tabControl.TabPages.Add(String.Format(CultureInfo.CurrentUICulture, "Tab {0}", _tabPageCount)).TabButtonImage = Resources.Blank
    End Sub

    Private Sub MainForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.AddTabPage()
    End Sub

    Private Sub _addTabButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _addTabButton.Click
        Me.AddTabPage()
    End Sub

    Dim _tabPageCount As Int32

    Sub New()
        Me.InitializeComponent()
    End Sub
End Class

File: ANGEN/SampleFramework.Controls/CommandManager/Samples/Tabs Category/Firefox TabControl/CS/FirefoxTabControl/MainForm.cs
using res = FirefoxTabControl.Properties.Resources;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Globalization;
using System.Windows.Forms;

namespace FirefoxTabControl
{
	public partial class MainForm : Form
	{
		protected override void OnLoad(EventArgs e)
		{
			base.OnLoad(e);
			this.AddTabPage();
		}

		private void AddTabPage()
		{
			_tabControl.TabPages.Add(String.Format(CultureInfo.CurrentUICulture, "Tab {0}", ++_tabPageCount)).TabButtonImage = res.Blank;
		}

		private void _addTabButton_Click(object sender, EventArgs e)
		{
			this.AddTabPage();
		}

		private Int32 _tabPageCount;

		public MainForm()
		{
			this.InitializeComponent();
		}
	}
}
